<a href="https://colab.research.google.com/github/Akshay2224/Uber_Request_Data/blob/master/Uber_Supply_EDA_Rpt.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Uber_Supply_Demand_Gap

## Introduction

Uber Technologies, Inc. is an American multinational transportation network company offering services that include peer-to-peer ridesharing, ride service hailing, food delivery, and a bicycle-sharing system. The company is based in San Francisco and has operations in 785 metropolitan areas worldwide.

![alt text](http://www.brooklynvegan.com/files/2018/08/uber.jpg))

## Problems Faced

The problems faced by customers, these very issues also impact the business of Uber. If drivers cancel the request of riders or if cars are unavailable, Uber loses out on its revenue. Let’s hear more about such problems that Uber faces during its operations. We need to  decide to address the problem Uber is facing - driver cancellation and non-availability of cars leading to loss of potential revenue. 

## Problem Statement

The aim of analysis is to identify the root cause of the problem (i.e. cancellation and non-availability of cars) and recommend ways to improve the situation. As a result of your analysis, you should be able to present to the client the root cause(s) and possible hypotheses of the problem(s) and recommend ways to improve them.  

 

In [0]:
# Importing of Libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from datetime import datetime

In [0]:
#opening of file
df=pd.read_csv('https://github.com/Akshay2224/Uber_Request_Data/raw/master/Uber%20Request%20Data.csv')

In [4]:
#Lets Head towards looking the dataset
df.head(10)

Request id Pickup point  ...    Request timestamp       Drop timestamp
0         619      Airport  ...      11/7/2016 11:51      11/7/2016 13:00
1         867      Airport  ...      11/7/2016 17:57      11/7/2016 18:47
2        1807         City  ...       12/7/2016 9:17       12/7/2016 9:58
3        2532      Airport  ...      12/7/2016 21:08      12/7/2016 22:03
4        3112         City  ...  13-07-2016 08:33:16  13-07-2016 09:25:47
5        3879      Airport  ...  13-07-2016 21:57:28  13-07-2016 22:28:59
6        4270      Airport  ...  14-07-2016 06:15:32  14-07-2016 07:13:15
7        5510      Airport  ...  15-07-2016 05:11:52  15-07-2016 06:07:52
8        6248         City  ...  15-07-2016 17:57:27  15-07-2016 18:50:51
9         267         City  ...       11/7/2016 6:46       11/7/2016 7:25

[10 rows x 6 columns]

## Attribute Description

There are six attributes associated with each request made by a customer:

1. Request id: A unique identifier of the request.
2. Time of request: The date and time at which the customer made the trip request
3. Drop-off time: The drop-off date and time, in case the trip was completed 
4. Pick-up point: The point from which the request was made
5. Driver id: The unique identification number of the driver
6. Status of the request: The final status of the trip, that can be either completed, cancelled by the driver or no cars available

## Some Stats about the dataset

In [46]:
# To get a description about the dataset
df.describe()

Request id    Driver id
count  6745.000000  4095.000000
mean   3384.644922   149.501343
std    1955.099667    86.051994
min       1.000000     1.000000
25%    1691.000000    75.000000
50%    3387.000000   149.000000
75%    5080.000000   224.000000
max    6766.000000   300.000000

#### We can see that number of request id are 6745 where 6766 is highest number of id alloted whereas we can see that Maximum number of driver Id is 300 and we have total 4095 drivers.

In [5]:
#To see the number of rows and columns
df.shape

(6745, 6)

In [7]:
#To check the name of columns 
df.columns

Index(['Request id', 'Pickup point', 'Driver id', 'Status',
       'Request timestamp', 'Drop timestamp'],
      dtype='object')

### Data Quality